Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -345,11 +345,12 @@ class CombinatorialKalmanFilter {
     /// Broadcast the result_type
     using result_type = CombinatorialKalmanFilterResult<traj_t>;
 
-    /// The filter target surface
-    const Surface* filterTargetSurface = nullptr;
+    /// The filter target surface aborter
+    SurfaceReached filterTargetReached{std::numeric_limits<double>::lowest()};
 
-    /// The smoothing target surface
-    const Surface* smoothingTargetSurface = nullptr;
+    /// The smoothing target surface aborter
+    SurfaceReached smoothingTargetReached{
```

Comment:
What if we disable the smoothing? I mentioned that we probably have to externalize the CKF smoothing to be able to copy them out. If we turn off the smoothing, will it also not try to reach this surface?

Reply:
this should still do the same thing if not set, yes